### PR TITLE
clean bone matrice weights - shadow problem

### PR DESCRIPTION
**Pull request by MatuskaLaszlo**

### CODE REVIEW

File: src/Loading/babylon.sceneLoader.ts [hunk 1]
```diff
@@ -331,6 +409,7 @@
                     }
 
                     if (onSuccess) {
+                        SceneLoader._CleanMatricesIndices(scene.meshes,scene.skeletons);
```
Comment (deltakosh) [inline src/Loading/babylon.sceneLoader.ts]: why only if Onsuccess is defined? We should do it everytime

File: src/Loading/babylon.sceneLoader.ts [hunk 2]
```diff
@@ -189,6 +189,82 @@
             }
         }
 
+        private static _CleanMatricesIndices(meshes:AbstractMesh[], skeletons:Skeleton[]): void {
```
Comment (deltakosh) [inline src/Loading/babylon.sceneLoader.ts]: I would like to keep it in the geometry class.
After loading a geometry we can run the cleanup.

File: src/Loading/babylon.sceneLoader.ts [hunk 3]
```diff
@@ -189,6 +189,82 @@
             }
         }
 
+        private static _CleanMatricesIndices(meshes:AbstractMesh[], skeletons:Skeleton[]): void {
+            if (!SceneLoader._CleanBoneMatrixWeights) {
+                return;
+            }
+            for (var i = 0; i < skeletons.length; i++) {
+                var skeleton = skeletons[i];
+                var noInfluenceBone = new Bone("noInfluence", skeleton, null, Matrix.Identity(), Matrix.Identity());
```
Comment (deltakosh) [inline src/Loading/babylon.sceneLoader.ts]: this is done automatically by the skeleton when preparing the matrices:
https://github.com/BabylonJS/Babylon.js/blob/master/src/Bones/babylon.skeleton.ts#L241

Comment (MatuskaLaszlo) [reply src/Loading/babylon.sceneLoader.ts]: Ok so it is always there at skeleton.bones.length, cool!

**Comment by deltakosh** [comment]
up?

**Comment by MatuskaLaszlo** [comment]
Sorry I'm on a vacation in beautiful Nice, France, but I will continue in a few days! 😁

**Comment by MatuskaLaszlo** [comment]
I will send another PR with the changes sorry

**Pull request closed without merging**